administration/src/graphql/applications/rejectApplicationStatus.graphql

```diff
@@ -1,5 +1,9 @@
-mutation rejectApplicationStatus($applicationId: Int!, $rejectionMessage: String!) {
-  application: rejectApplicationStatus(applicationId: $applicationId, rejectionMessage: $rejectionMessage) {
+mutation rejectApplicationStatus($applicationId: Int!, $project: String!, $rejectionMessage: String!) {
+  application: rejectApplicationStatus(
+    applicationId: $applicationId
+    project: $project
```

Comment:
❓ I know this is used in a lot of GraphQL functions, but I cannot imagine this is a good idea to have this as a parameter. Shouln't this come from a build time constant or configuration somewhere along the deployment process? Wouldn't this allow any logged in user to issue any project name and mess with an unrelated project here?

Reply:
Perhaps I misunderstand your point, but we have one backend instance across all projects. So we can‘t have project as a constant in the backend.

Reply:
I don't have any insights into the deployment process yet, so I'm just speculating here. Are users then supposed to manipulate data across projects?

Reply:
Then shouldn't the project name be derived from the current sessions administrator `projectId` property instead of blindly trusting user input here?

This seems like a medium security issue here. It's not catastrophic, since users must be authenticated, but we have real possibility for unauthorized access across projects here.

Or did I miss something here?

Reply:
I think it's a cool observation.
If we take the project from the admin - it would mean an additional database request to get the project name by its id, but it's not a big deal, I guess.

Reply:
Yes we may fix that. The only thing that can happen in this case is that the mail signature is from a different project but this would be prevented by the `Authroizer` so no panic :)

But we will have cases where we have to pass the project because the request doesn't enforce authorization